frontend/src/components/table/TableCell.js

```diff
@@ -1,184 +1,51 @@
-          return fieldValue
-            .map((value) => TableCell.fieldValueToString(value, fieldType))
-            .join(' - ');
-        }
-
-        return DATE_FIELD_TYPES.includes(fieldType) ||
-          TIME_FIELD_TYPES.includes(fieldType)
-          ? TableCell.createDate(fieldValue, fieldType)
-          : fieldValue.caption;
-      }
-      case 'boolean': {
-        return fieldValue ? (
-          <i className="meta-icon-checkbox-1" />
-        ) : (
-          <i className="meta-icon-checkbox" />
-        );
-      }
-      case 'string': {
-        if (
-          DATE_FIELD_TYPES.includes(fieldType) ||
-          TIME_FIELD_TYPES.includes(fieldType)
-        ) {
-          return TableCell.createDate(fieldValue, fieldType);
-        } else if (AMOUNT_FIELD_TYPES.includes(fieldType)) {
-          return TableCell.createAmount(fieldValue, precision, isGerman);
-        } else if (SPECIAL_FIELD_TYPES.includes(fieldType)) {
-          return TableCell.createSpecialField(fieldType, fieldValue);
-        }
-        return fieldValue;
-      }
-      default: {
-        return fieldValue;
-      }
-    }
-  };
-
   constructor(props) {
     super(props);
-
     this.widget = createRef();
-
     this.clearWidgetValue = false;
 
     this.state = {
-      tooltipToggled: false,
+      tooltipToggled: false, // keeping in the local state the flag for the tooltip
     };
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    const { updateRow, readonly, rowId, tdValue } = this.props;
-    const { tdValue: nextTdValue } = nextProps;
-    // We should avoid highlighting when whole row is exchanged (sorting)
-    if (rowId !== nextProps.rowId) {
-      return;
-    }
-
-    if (!readonly && tdValue !== nextTdValue) {
-      updateRow();
-    }
-  }
-
-  widgetTooltipToggle = (field, value) => {
+  /**
+   * @method widgetTooltipToggle
+   * @summary Alternative method to open dropdown, in case of disabled opening on focus.
+   * @param {string|null} value
+   */
+  widgetTooltipToggle = (value) => {
     const curVal = this.state.tooltipToggled;
     const newVal = value != null ? value : !curVal;
 
-    this.setState({
-      tooltipToggled: newVal,
-    });
+    this.setState({ tooltipToggled: newVal });
   };
 
+  /**
+   * @method handleBackdropLock
+   * @summary checks widget against widget list and calls parent onClickOutside fnct
+   * @param {object} state
+   */
   handleBackdropLock = (state) => {
     const { item } = this.props;
-
-    if (
-      !['ProductAttributes', 'Attributes', 'List', 'Lookup'].includes(
-        item.widgetType
-      )
-    ) {
-      if (!state) {
-        this.props.onClickOutside();
-      }
+    const widgetsList = ['ProductAttributes', 'Attributes', 'List', 'Lookup'];
+    if (!widgetsList.includes(item.widgetType)) {
+      !state && this.props.onClickOutside();
     }
   };
 
+  /**
+   * @method handleKeyDown
+   * @summary Key down function handler
+   * @param {object} e
```

Comment:
is there a type `event` ?

Reply:
this is the corresponding event from a text input for example when you change a value within a table cell by typing something in that specific cell.

Reply:
@petrican  why not writing this info as `@param e` documentation?

Reply:
Ok I'll add it in there. Thanks